daal4py/sklearn/ensemble/_forest.py

```diff
@@ -704,32 +704,19 @@ def predict_proba(self, X):
             The class probabilities of the input samples. The order of the
             classes corresponds to that in the attribute :term:`classes_`.
         """
-        # Temporary solution
-        X = check_array(X, accept_sparse=['csr', 'csc', 'coo'])
-        if hasattr(self, 'n_features_in_'):
-            if X.shape[1] != self.n_features_in_:
-                raise ValueError(
-                    (f'X has {X.shape[1]} features, '
-                     f'but RandomForestClassifier is expecting '
-                     f'{self.n_features_in_} features as input'))
+        X = check_array(X, accept_sparse=['csr', 'csc', 'coo'],

```

Comment:
move this after check for use daal, otherwise will do unnecessary checks

Reply:
Ok